test/jdk/sun/security/rsa/SignatureTest.java

```diff
@@ -133,12 +135,18 @@ private static KeyPair generateKeys(String keyalg, int size)
                 } catch (InvalidKeySpecException expected) {
                 }
 
+                RSAPublicKeySpec pubKeySpec1 = kf.getKeySpec(key, RSAPublicKeySpec.class);
+                RSAPublicKeySpec pubKeySpec2 = new RSAPublicKeySpec(
+                        ((RSAPublicKey) key).getModulus(),
+                        ((RSAPublicKey) key).getPublicExponent());
+
+                Asserts.assertTrue(keySpecEquals(pubKeySpec1, pubKeySpec2),
+                        "Both RSAPublicKeySpec should be equal");
+
```

Comment:
Do you also want to add similar check on X509EncodedKeySpec? Same for PKCS8EncodedKeySpec below.

Reply:
Both verifications included now